packages/uploadthing/src/server.ts

```diff
@@ -24,73 +20,16 @@ export const createUploadthing = <TErrorShape extends Json>(
   opts?: CreateBuilderOptions<TErrorShape>,
 ) => createBuilder<MiddlewareArgs, TErrorShape>(opts);
 
-export interface ResponseWithCleanup extends Response {
-  /** custom property where a Promise may be put that you can await in for example Cloudflare Workers */
-  cleanup?: Promise<unknown>;
-}
-
-/** @internal */
-export const INTERNAL_DO_NOT_USE_createRouteHandlerCore = <
-  TRouter extends FileRouter,
->(
+export const createRouteHandler = <TRouter extends FileRouter>(
   opts: RouteHandlerOptions<TRouter>,
-  adapter: string,
 ) => {
-  incompatibleNodeGuard();
-
-  const requestHandler = buildRequestHandler<TRouter, MiddlewareArgs>(
+  return makeAdapterHandler<[Request]>(
+    (req) => Effect.succeed({ req, res: undefined, event: undefined }),
+    (req) => Effect.succeed(req),
     opts,
-    adapter,
+    "server",
   );
```

Comment:
intuitively it felt like it makes sense to have the "vanilla" server handler not care about the method (that's all handled internally), so I changed the server handler to be

```ts
(req: Request) => Promise<Response>
```

instead of 
```ts
{
  GET: (req: Request) => Promise<Response>,
  POST: (req: Request) => Promise<Response>
}
```

what do you think?